### PR TITLE
WIP: Support xstream

### PR DESCRIPTION
**Pull request by Widdershin**
I initially tried a fresh rebuild with recycle, but it's not worth it to lose the tests and bug fixes.

Instead I am going to try updating this, and then refactoring to take some of the improvements from recycle.

### CODE REVIEW

**Comment by Widdershin** [comment]
I think this will work for the super basic DOM case once the DOM disposal fix is shipped.

https://github.com/cyclejs/cyclejs/pull/442

That or when Frikki's refactor is done.

**Comment by Widdershin** [comment]
You can use xdescribe to mark a block as pending. Deliberate :)

**Comment by TylorS** [comment]
The more you know :D 

Thanks!

**Comment by Widdershin** [comment]
Blocked awaiting https://github.com/cyclejs/cyclejs/issues/476 to be fixed.

DOM driver is working, HTTP driver works when the hack fix for 476 is manually applied, but I'm not comfortable shipping until it works out of the box.

**Comment by TylorS** [comment]
Awesome work @Widdershin. Will try to prioritize a fix for that issue

File: package.json [hunk 1]
```diff
@@ -34,11 +34,11 @@
   },
   "homepage": "https://github.com/Widdershin/cycle-restart",
   "devDependencies": {
-    "@cycle/dom": "^9.0.1",
     "@cycle/history": "^1.1.0",
-    "@cycle/http": "^7.1.0",
-    "@cycle/isolate": "^1.2.0",
+    "@cycle/http": "^11.2.0",
```
Comment (Widdershin) [inline package.json]: Need to purge some dependencies

File: src/restart.js [hunk 2]
```diff
@@ -18,7 +18,15 @@ function restart (main, drivers, {sources, sinks}, isolate = {}, timeToTravelTo
...
+  const newDispose = newSourcesSinksAndRun.run();
+
+  const sourcesAndSinksAndDispose = {
+    sources: newSourcesSinksAndRun.sources,
+    sinks: newSourcesSinksAndRun.sinks,
+    dispose: newDispose
+  };
 
   setTimeout(() => {
     const scheduler = new Rx.HistoricalScheduler();
```
Comment (Widdershin) [inline src/restart.js]: Once `@cycle/time` is released it can replace this scheduler.

File: src/restartable.js [hunk 3]
```diff
@@ -117,56 +135,90 @@ function wrapSource ({streams, addLogEntry, pause$}, source, scope = []) {
   return returnValue;
 }
 
+const shittyListener = {
```
Comment (Widdershin) [inline src/restartable.js]: Really need to handle errors in these.

Also need a better name

**Pull request merged**